apiserver/facades/client/application/application.go

```diff
@@ -457,9 +457,19 @@ func (api *APIBase) Deploy(args params.ApplicationsDeploy) (params.ErrorResults,
 // the params.ApplicationsDeploy changed. Thus it appears you can use
 // a juju 3.x client to deploy from a juju 2.9 controller, however it
 // fails because the series was not found. Make those corrections here.
+//
+// However, we still need to support some clients which do not pass OS
+// data in the origin, only the param series/charmurl. If a series is found,
+// in either the origin or arg.Series, return early. origin.Series takes
+// priority
 func compatibilityApplicationDeployArgs(arg params.ApplicationDeploy) (params.ApplicationDeploy, error) {
 	origin := arg.CharmOrigin
 	if origin.Series != "" {
+		arg.Series = origin.Series
+		return arg, nil
+	}
+	if arg.Series != "" {
+		origin.Series = arg.Series
```

Comment:
It's not guarenteed, no. In fact, I believe both will be blank always for 3.x juju CLI clients

In this case, we get the OS from the origin base

If there is no arg.Series, origin.Series or origin.Base we will get an error on what is now line 477, since the parse will fail.

Granted the message will be "missing base name or channel not valid" which is pretty bad